execution-engine/src/hcall/common.rs

```diff
@@ -275,578 +160,57 @@ impl<T> From<std::sync::PoisonError<T>> for HostProvisioningError {
-                expected: vec![LifecycleState::StreamSourcesLoading],
-                found: otherwise.clone(),
-            }),
-        }
-    }
-
-    /// Sort the incoming data against pattern, against the expected_data_sources provided in the policy file
-    fn sorting_functor(
-        data: &mut Vec<DataSourceMetadata>,
-        pattern: &Vec<u64>,
-    ) -> Result<(), HostProvisioningError> {
-        // pre-sort the incoming data by client id then package id
-        data.sort_by(|lhs, rhs| {
-            lhs.client_id
-                .cmp(&rhs.client_id)
-                .then_with(|| lhs.package_id.cmp(&rhs.package_id))
-        });
-
-        // Iterate over pattern.
-        // In each iteration, search the next package in `data` and add the result to
-        // `new_data_sources`. Variable `previous_packages` tracks the previous package_id for each
-        // client, initially being undefined and then updated in each iteration.
-        let mut new_data_sources: std::vec::Vec<DataSourceMetadata> = Vec::new();
-        let mut previous_packages: std::collections::HashMap<u64, u64> = HashMap::new();
-        for next_client_id in pattern.iter() {
-            let next_package_iter = match previous_packages.get(&next_client_id) {
-                Some(previous_package_id) => data.iter().find(|data_item| {
-                    data_item.client_id == *next_client_id
-                        && data_item.package_id > *previous_package_id
-                }),
-                None => {
-                    // because the data_sources is sorted, the first package in the vec for client_id will be the first package
-                    data.iter()
-                        .find(|data_item| data_item.client_id == *next_client_id)
-                }
-            };
-
-            match next_package_iter {
-                Some(next_package) => {
-                    previous_packages.insert(*next_client_id, next_package.package_id);
-
-                    new_data_sources.push(DataSourceMetadata {
-                        data: next_package.data.iter().cloned().collect(),
-                        client_id: next_package.client_id,
-                        package_id: next_package.package_id,
-                    });
-                }
-                None => {
-                    return Err(HostProvisioningError::CannotSortDataOrStream);
-                }
-            };
-        }
-        *data = new_data_sources;
-        Ok(())
-    }
-
-    ////////////////////////////////////////////////////////////////////////////
-    // Requesting shutdown.
-    ////////////////////////////////////////////////////////////////////////////
-
-    /// Signals to the provisioning host that a client has requested shutdown.
-    #[inline]
-    pub(crate) fn request_shutdown(&mut self, client_id: u64) {
-        self.expected_shutdown_sources.retain(|v| v != &client_id);
+    /// Read from a file
```

Comment:
Same comment: is this really a file read?

Reply:
Fixed.

Reply:
The Rustdoc comment still seems to be incorrect?